gammapy/cube/core.py

```diff
@@ -233,24 +249,81 @@ def empty(cls, emin=0.5, emax=100, enumbins=10, eunit='TeV', mode='edges', **kwa
         return cls(data=data, wcs=image.wcs, energy_axis=energy_axis)
 
     @classmethod
-    def empty_like(cls, refcube, fill=0, unit=''):
+    def empty_like(cls, reference, energies=None, fill=0):
         """
-        Create an empty sky cube with the same WCS, energy specification and meta
-        as given sky cube.
+        Create an empty sky cube with the same WCS and energy specification
+        as given reference.
 
         Parameters
         ----------
-        refcube : `~gammapy.cube.SkyCube`
-            Reference sky cube.
-        fill : float, optional
-            Fill image with constant value. Default is 0.
-        unit : str or `~astropy.units.Unit`
-            Unit of the data for the empty cube.
-        """
-        wcs = refcube.wcs.copy()
-        data = fill * np.ones_like(refcube.data) * u.Unit(unit)
-        energy_axis = refcube.energy_axis
-        return cls(data=data, wcs=wcs, energy_axis=energy_axis, meta=refcube.meta)
+        reference : `~gammapy.cube.SkyCube` or `~gammapy.image.SkyImage`
+            Reference sky cube or image.
+        energies : `~gammapy.utils.Energy` or `~gammapy.utils.EnergyBounds` (optional)
+            Reference energies, mandatory when a `~gammapy.image.SkyImage` is passed.
+        fill : float
+            Value to fill the data array with.
+
+        Examples
+        --------
+        Create an empty sky cube from an image and energy center specification:
+        
+        .. code::
+
+            from astropy import units as u
+            from gammapy.image import SkyImage
+            from gammapy.cube import SkyCube
+            from gammapy.utils.energy import Energy
+
+            image = SkyImage.empty(nxpix=11, nypix=7)
+            energies = Energy.equal_log_spacing(1 * u.TeV, 100 * u.TeV, 3)
+            cube = SkyCube.empty_like(reference=image, energies=energies)
+
+        Create an empty sky cube from an image and energy bounds specification:
+
+        .. code::
+
+            from astropy import units as u
+            from gammapy.image import SkyImage
+            from gammapy.cube import SkyCube
+            from gammapy.utils.energy import EnergyBounds
+
+            image = SkyImage.empty(nxpix=11, nypix=7)
+            energies = EnergyBounds.equal_log_spacing(1 * u.TeV, 100 * u.TeV, 3)
+            cube = SkyCube.empty_like(reference=image, energies=energies)
```

Comment:
The two code examples could be combined (just separated by empty line), to avoid repeating the same 4 input lines. Not sure if it's better, just a thought.